tsdb/db.go

```diff
@@ -611,8 +611,10 @@ func (db *DB) run() {
 			}
 			db.autoCompactMtx.Unlock()
 		case <-db.stopc:
+			timer.Stop()
```

Comment:
This could be in a `defer` statement just after initialization.

Reply:
defer will hold a reference for each timer in that case and that will be a memory leak.